### ISSUE TITLE
a error of creating Azure Logic App with Terraform

### ISSUE THREAD

**Issue by cuntoulishifu** [terraform-providers/terraform-provider-azurerm#7589]
Hi Host,

I am creating Azure Logic App with Terraform through below document  https://www.terraform.io/docs/providers/azurerm/r/logic_app_workflow.html

but I encountered an error as blow:
```hcl
Error creating Logic App Workflow "workflow1" (Resource Group "Bo_Li"): logic.WorkflowsClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400
Code="NoRegisteredProviderFound" Message="No registered resource provider found for location 'chinaeast' and API version '2019-05-01' for type 'workflows'. The supported api-versions are '2015-02-01-preview, 2015-08-01-preview, 2016-06-01, 2016-10-01, 2017-07-01, 2018-07-01-preview'. The supported locations are 'chinanorth, chinaeast'."
```
I show my Terraform environment in there
```hcl
Terraform v0.12.21
provider.azurerm v2.13.0
terraform-provider-azurerm_v2.13.0_x5.exe
```
I‘d like to know what can I do for that? looking forward to your reply, Thanks a lot!

**Comment by tombuildsstuff** [comment]
Hi @cuntoulishifu 

Thanks for opening this issue.

Based on the error message being returned from the Azure API here, it appears that the versions of the Logic Apps API deployed within Azure China are older than the versions available in Azure Public - where we're using version `2019-05-01`.

Since the newer version of the Azure API we require isn't available within Azure China, it should be possible to use an older version of the Azure Provider to provision Logic Apps in Azure China (which targets the older API).

Based on what I can see the SDK upgrade was introduced in #5862, which shipped in v2.2.0 of the Azure Provider - as such you'd need to pin to v2.1.0 for this to work within Azure China, for example:

```
provider "azurerm" {
  version = "=2.1.0"
  features {}
}
```

At which point running `terraform init` will download this version of the Azure Provider and you should be able to provision this Logic App Workflow. Alternatively you deploy this Logic App Workflow using an ARM Template Deployment, via the `azurerm_template_deployment` resource.

Since this should be fixed by updating the Terraform Configuration being used here (either to pin to an older provider version, or to use the ARM Template Deployment resource to provision this) - I'm going to close this issue for the moment, but please let us know if that doesn't work for you and we'll take another look.

Thanks!

**Comment by cuntoulishifu** [comment]
@tombuildsstuff  Thanks very much! Terraform works fine now!
